### DISCUSSION TITLE
NFT sync gaming API for NEAR ecosystem

### DISCUSSION THREAD

**Discussion by jilt** [filecoin-project/community#476] [Project Showcase]
This project builds on top of an unlockable content service built for NEAR nfts and relying on web3.storage as an immutable DB service (list() function, basically).
The unlockable content platform will be launched by the end of march, you can see an alpha version at [this link](https://vault.varda.vision/) and I've been asked by some game devs to be able to verify ownership for their NFTs in game using an API.
I often speak to game devs because I'm part of the [humaguild DAO](https://app.astrodao.com/dao/human.sputnik-dao.near/proposals) and distribute funding on near, so I'm sure this NFT gaming API could leverage lots of users in the near gaming ecosystem.
The API would consist of dedicated files and subgraphs (to update ownership) for each game/smart contract.

**Comment by jilt** [comment]
The pull request has changed! [here is the link](https://github.com/filecoin-project/community/pull/488)

**Comment by jilt** [comment]
Hi guys,
I'm a little confused here, been receiving emails saying that i didn't submit the form for the grant in the right way, but i did, and also got the tax document checked by email... I know there's a deadline for the funding to arrive and was wondering why i keep attaching the checked file to your google form but get emails saying i didn't attach the file, do you guys want it to be attached here? Info in thatis quite sensitive... should i share a screenshot proving I actually have attached the file to the form?

**Comment by jilt** [comment]
Hi people, I decided to create a [new template repository](https://jilt.github.io/filecoin-near-upload-app/), as [the one i created before](https://github.com/jilt/near-create-app-webpack5-bundle) and referenced on the microgrants was too generic and didn't address [a problem with the npm package imports](https://github.com/web3-storage/web3.storage/issues/1301), happening while uploading files with the javascript client from web3.storage.

My template solves this problem and is out with MIT license so everyone can Use it on the next hackathons (ending soon afaik).

The [Gaming API is Up and running](https://github.com/jilt/Vault-API-Filecoin), and the main Varda Vault interface will be soon updated to reference it.
By now the swagger generated docs are [here](https://jilt.github.io/Vault-API-Filecoin/) if anyone wanna build a simple NFT game on NEAR, minting on [Paras](https://paras.id/) or [Mintbase](https://www.mintbase.io/) you can use our unlockable files endpoint to refence a 3D model for your metaverse or any extra asset to your NFT art.

Will message again when the Vault interface will be updated!
Thanks guys, hope your community will make good use of this!

**Reply by jilt** [reply]
File upload app demo [link](https://jilt.github.io/filecoin-near-upload-app/)

**Comment by jilt** [comment]
Hi people, I've been asked by a gaming project on near if using this API and nft.storage can make their project eligible for a microgrant, what should i tell them?

**Reply by web3jenks** [reply]
Hi @jilt , NFT.storage is one of the Filecoin storage helpers, so we will consider it eligible. However, a decision on a grant is not solely based on whether they are using the tech. There is a range of considerations that Grants team will use to determine a grant approval. :-) Please do encourage the project to apply for our micro/open grant.
